### PR TITLE
feat: support credStore

### PR DESCRIPTION
**Pull request by thesayyn**
PR also introduces auth/token caching which is needed otherwise the credential helpers are called repetitively which is not suboptimal. 

I've done my best to keep changes small.

### CODE REVIEW

File: oci/private/auth_config_locator.bzl [hunk 1]
```diff
@@ -41,7 +41,6 @@ def _get_auth_file_path(rctx):
 
 def _oci_auth_config_locator_impl(rctx):
     config_path = _get_auth_file_path(rctx)
-    config_path = None
```
Comment (thesayyn) [inline oci/private/auth_config_locator.bzl:44]: https://github.com/bazel-contrib/rules_oci/pull/150

File: oci/pull.bzl [hunk 2]
```diff
@@ -218,6 +271,17 @@ Falling back to using `curl`. See https://github.com/bazelbuild/bazel/issues/178
 
     return manifest, len(bytes)
 
+def _create_downloader(rctx):
+    state = {
+        "config": json.decode(rctx.read(rctx.attr.config)),
```
Comment (alexeagle) [inline oci/pull.bzl:276]: should there be some error handling in case you point us to a wrong file or it isn't valid json?

**Review by alexeagle — APPROVED** [review]
I didn't read through the new e2e but I'm happy it's there

File: oci/pull.bzl (hunk 2 above)
Comment (thesayyn) [reply oci/pull.bzl:276]: json.decode throws in invalid json case. for other cases, we check if the key we are looking for exists in the json file. so no for now.

**Pull request merged**